lib/rubygems/specification.rb

```diff
@@ -1153,7 +1153,7 @@ def self.non_nil_attributes
 
   def self.normalize_yaml_input(input)
     result = input.respond_to?(:read) ? input.read : input
-    result = "--- " + result unless result =~ /\A--- /
+    result = "--- " + result unless result.start_with?("--- ")
```

Comment:
This could potentially explode on `nil` input as well (called from `from_yaml`, which will explode later anyway on `nil` input).

Reply:
I checked this one and it can't be `nil` since `gsub!` is being called on it two lines later.

Reply:
I mean, we're not introducing any issue that's not already there, and since we haven't had any crash reports here, I think we're fine.